modules/prebid/ortb2blocking/hook_bidderrequest.go

```diff
@@ -0,0 +1,388 @@
+package ortb2blocking
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/prebid/openrtb/v17/adcom1"
+	"github.com/prebid/openrtb/v17/openrtb2"
+	"github.com/prebid/prebid-server/hooks/hookexecution"
+	"github.com/prebid/prebid-server/hooks/hookstage"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+func handleBidderRequestHook(
+	cfg Config,
+	payload hookstage.BidderRequestPayload,
+) (result hookstage.HookResult[hookstage.BidderRequestPayload], err error) {
+	if payload.BidRequest == nil {
+		return result, hookexecution.NewFailure("empty BidRequest provided")
+	}
+
+	bidder := payload.Bidder
+	mediaTypes := mediaTypesFrom(payload.BidRequest)
+	changeSet := hookstage.ChangeSet[hookstage.BidderRequestPayload]{}
+	blockingAttributes := blockingAttributes{}
+
+	if err = updateBAdv(cfg, bidder, mediaTypes, &blockingAttributes, &result, &changeSet); err != nil {
+		return result, hookexecution.NewFailure("failed to update badv field: %s", err)
+	}
+
+	if err = updateBApp(cfg, bidder, mediaTypes, &blockingAttributes, &result, &changeSet); err != nil {
+		return result, hookexecution.NewFailure("failed to update bapp field: %s", err)
+	}
+
+	if err = updateBCat(cfg, bidder, mediaTypes, &blockingAttributes, &result, &changeSet); err != nil {
+		return result, hookexecution.NewFailure("failed to update bcat field: %s", err)
+	}
+
+	if err = updateBType(cfg, payload, &blockingAttributes, &result, &changeSet); err != nil {
+		return result, hookexecution.NewFailure("failed to update btype field: %s", err)
+	}
+
+	if err = updateBAttr(cfg, payload, &blockingAttributes, &result, &changeSet); err != nil {
+		return result, hookexecution.NewFailure("failed to update battr field: %s", err)
+	}
+
+	// todo: probably we should update cattax only if current bidder supports openrtb v2.6
+	blockingAttributes.cattax = payload.BidRequest.CatTax
```

Comment:
First of all, I really like how clean this method is. Thanks :) Can we please also move the logic related to category taxonomy in a separate function so that this function is just calling update function for all these diff fields and doesn't have specific logic to any of them here?

Reply:
Of course, updated ✔️